### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by hamidazim321**
# PR Summary
- We have fixed the N+1 query problem by eager loading posts and comments for a user in the posts controller:
     - **before:**
posts_before
     - **after: comments and posts are loaded at the same time as user**
posts_after
- We have added feature tests for both posts and users for all features required in the project

## Changes made after previous review
I have added tests for all requested changes tests

- Tests that assert you can see the first comments on a post.
- Tests that assert you can see how many comments a post has.
- Tests that assert you can see how many likes a post has.
- Tests that assert you can see a section for pagination if there are more posts than fit on the view.

### CODE REVIEW

File: spec/features/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  context 'Click' do
+    it "redirects me to that post's show page when I click on a post" do
+      visit user_posts_path(@user)
+      if @user.posts.any?
+        recent_post = @user.posts[0]
+        click_link recent_post.title
+        expect(page).to have_current_path(user_post_path(@user, recent_post))
+      end
+    end
+  end
```
Comment (Ibizugbe) [inline spec/features/posts_index_spec.rb]: Nice job so far👋🏼👋🏼. Currently, your post_index_spec lacks the following tests😉
- Tests that assert you can see the first comments on a post.
- Tests that assert you can see how many comments a post has.
- Tests that assert you can see how many likes a post has.
- Tests that assert you can see a section for pagination if there are more posts than fit on the view.

Please kindly do well to add them here as they are [required](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md) in this milestone😉.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @hamidazim321  @DevKaranJ 👋🏼👋🏼👋🏼, 

request change


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] Gemfile has capybara gem👍🏼.
- [x] All tests are currently passing👍🏼.
- [x] User index and show page are properly tested👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @hamidazim321, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 




## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @hamidazim321,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### This project has been approved. You need to repeat this project on your other partner's (Karan Joshi) repo and have the repo owner submit from their dashboard 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**